runtime/autoload/man.vim

```diff
@@ -309,17 +307,54 @@ function! s:format_candidate(path, psect) abort
   endif
 endfunction
 
+function! s:init_highlight_groups() abort
+  highlight default manBold cterm=bold gui=bold
+  highlight default manUnderline cterm=underline gui=underline
+endfunction
+augroup man_colorscheme
+ autocmd!
+ autocmd ColorScheme * call s:init_highlight_groups()
+augroup END
+call s:init_highlight_groups()
+
+function! s:strip_backspaced_text(match) abort
+  let s:stripped = substitute(a:match, '.\b', '', 'g')
+  return s:stripped
+endfunction
+
+let s:src_id = nvim_buf_add_highlight(0, 0, '', 0, 0, 0)
+function! man#highlight_backspaced_text() abort
+  call nvim_buf_clear_highlight(0, s:src_id, 0, -1)
+  while 1
+    let pos = searchpos('\%(_\b[^_]\)\|\%(\(.\)\b\1\)', 'p')
```

Comment:
`searchpos()` has a "timeout" parameter, but it's not very useful in a while-loop. And calling `reltime()` in a loop will be too expensive. So maybe the loop should be limited to 1000 iterations.